Automattic/wp-calypso
# Stats: Add upgrade buttons for already purchase notices

## Proposed Changes

This is a follow up for https://github.com/Automattic/wp-calypso/pull/81250.

The PR proposes to add 'Upgrade my Stats' for the already-purchased notices. Free would be directed to PWYW interface. Personal would be directed to commercial purchase page.

## Testing Instructions

* Open Calypso Live
* Remove any Stats products for a site
* Open `/stats/purchase/:siteId` and purchase Free for the site
* Open `/stats/purchase/:siteId?flags=stats/type-detection`
* Ensure you see the following screen
* Click Upgrade my Stats
* Ensure you are taken to PWYW purchase screen (See your Stats was change to secondary to boost attention on upgrade)

<img width="623" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1425433/7d5f6a45-43c9-40d6-85af-e0188be4952d">


* Purchase a Personal plan
* Open `/stats/purchase/:siteId?flags=stats/type-detection`.
* Click Upgrade my Stats
* Ensure you are taken to Commercial purchase screen (See your Stats was change to secondary to boost attention on upgrade)

<img width="641" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1425433/14080dc7-9905-4820-a73d-da864f70337b">


<img width="607" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1425433/2b769b50-ea1e-4320-86e6-1faa7e8f905a">



* Ensure you can purchase a Commercial plan

## Pre-merge Checklist
